Extension / Rule change will not invalidate the cache

marcelthole opened the pull request:
Hey 👋 
we have the problem that the cache file is not invalidated if we change the code inside of a custom rule.

```neon
services:
    -
        class: CustomPHPStanRule\CustomRuleA
parameters:
	inferPrivatePropertyTypeFromConstructor: true
```

```bash
$ cat cache/phpstan/resultCache.php  | grep -A1 "projectExtensionFiles"
        'projectExtensionFiles' => array (
        ),
```

I expected the CustomRuleA in the projectExtensionFile. if this is there the normal cache workflow will also check the sha1 file sum and invalidates the cache.

With the current change the current service is always added as projectExtensionFile and every change on that file will invalidate the cache

marcelthole committed:
Add all passed services as projectExtensionFiles even if they don't have dependencies to invalidate the search cache on changes inside of the service

ondrejmirtes commented:
Please try to reproduce the problem first. Here you can see how we typically test result cache-related problems https://github.com/phpstan/phpstan-src/blob/1.11.x/.github/workflows/e2e-tests.yml

marcelthole committed:
Add failing testcase for cached custom rules
If a custom rule in a project will change, this must invalidate the cache to allow the new rule to find new issues

ondrejmirtes commented on e2e/result-cache-8/CustomRule.php:
```diff
@@ -0,0 +1,21 @@
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+
+class CustomRule implements Rule
```
Your custom rule should be part of your analysed paths. If you fix that I don't think you'll have this problem anymore.

marcelthole replied:
> Result cache not used because extension file /var/www/e2e/result-cache-8/src/CustomRule.php hash does not match.

👍 nice thanks a lot for the hint
